OpenDreamServer/Dream/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -1231,5 +1233,28 @@ public static DreamValue NativeProc_viewers(DreamObject instance, DreamObject us
 
             return DreamValue.Null;
         }
+        
+        [DreamProc("md5")]
+        [DreamProcParameter("T", Type = DreamValueType.String | DreamValueType.DreamResource)]
+        public static DreamValue NativeProc_md5(DreamObject instance, DreamObject usr, DreamProcArguments arguments) {
+            if(arguments.ArgumentCount > 1) throw new Exception("md5() only takes one argument");
+            DreamValue arg = arguments.GetArgument(0, "T");
+            DreamResource resource;
+
+            if (arg.Type == DreamValueType.String) {
+                resource = Program.DreamResourceManager.LoadResource(arg.GetValueAsString());
```

Comment:
Giving `md5()` a string always hashes the string, it doesn't attempt to load it as a file.